lua/neogit/popups/branch/actions.lua

```diff
@@ -22,6 +22,40 @@ local function parse_remote_branch_name(ref)
   return remote, branch_name
 end
 
+M.spin_off_branch = operation("spin_off_branch", function()
+  if #git.repo.staged.items > 0 or #git.repo.unstaged.items > 0 then
+    notif.create("Staying on current branch as there are uncommitted changes.", vim.log.levels.INFO)
```

Comment:
Looking at how Magit handles this (https://github.com/magit/magit/blob/main/lisp/magit-branch.el#L490-L492), this guard clause only applies if we are _not_ checking out the new branch, ie, for `spin_out` but not for `spin_off`

Reply:
Ah, you're right 👍🏼

I think I misunderstood this part entirely. If there are uncommitted changes, Magit doesn't short-circuit the operation, it [forces checkout](https://github.com/magit/magit/blob/main/lisp/magit-branch.el#L493) so the changes aren't lost. This behaviour is actually called out in the docs but didn't make sense to me until now:
> If there are any uncommitted changes, then behave exactly like `magit-branch-spinoff'.

Given the above, I'll:
- Remove this guard from `spin_off`
- Make `spin_out` behave like `spin_off` if there are uncommitted changes

Reply:
No worries, I had misread it too (in the opposite direction). Sounds like you're on the right track now :D